FlowCrypt/Functionality/Services/Organisational Rules Service/OrganisationalRulesService.swift

```diff
@@ -67,6 +72,32 @@ extension OrganisationalRulesService: OrganisationalRulesServiceType {
         }
     }
 
+    func getEmailKeyManagerPrivateKeys() -> Promise<GetEKMKeysResult> {
+        Promise<GetEKMKeysResult> { [weak self] resolve, _ in
+            guard let self = self else { throw AppErr.nilSelf }
+            let organisationalRules = self.getSavedOrganisationalRulesForCurrentUser()
+            guard let keyManagerUrlString = organisationalRules.keyManagerUrlString else {
+                resolve(([], nil))
+                return
+            }
+            let urlString = "\(keyManagerUrlString)v1/keys/private"
+            let headers = [
+                URLHeader(
+                    value: "Bearer \(GoogleUserService().idToken ?? "")",
+                    httpHeaderField: "Authorization"
+                )]
+            let request = URLRequest.urlRequest(
+                with: urlString,
+                method: .get,
+                body: nil,
+                headers: headers
+            )
+            let response = try awaitPromise(URLSession.shared.call(request))
+            let container = try JSONDecoder().decode(DecryptedPrivateKeysContainer.self, from: response.data)
+            resolve((container.privateKeys, urlString))
```

Comment:
Also here, let's not alter this response just to return the EKM url, better retrieve EKM url again with `organisationalRules.keyManagerUrlString` wherever this is used

Reply:
I am passing it to show url in toast, I am sure that we should make another request just to put url into a toast.